util/helm/cmd.go

```diff
@@ -342,19 +342,32 @@ type TemplateOpts struct {
 	SkipCrds    bool
 }
 
-var (
-	re                 = regexp.MustCompile(`([^\\]),`)
-	apiVersionsRemover = regexp.MustCompile(`(--api-versions [^ ]+ )+`)
-)
-
 func cleanSetParameters(val string) string {
 	// `{}` equal helm list parameters format, so don't escape `,`.
 	if strings.HasPrefix(val, `{`) && strings.HasSuffix(val, `}`) {
 		return val
 	}
-	return re.ReplaceAllString(val, `$1\,`)
+
+	var result strings.Builder
+	var prevR rune
+	for _, r := range val {
+		if r == ',' {
+			if prevR != '\\' {
+				result.WriteString(`\,`)
+			} else {
+				result.WriteRune(',')
+			}
+		} else {
+			result.WriteRune(r)
+		}
+		prevR = r
+	}
+
+	return result.String()
```

Comment:
Extract to a function 😄  
In the future, this will be hard to understand what it does. 

Something like`replaceAllWithLookbehind(s, old, new String, lookbehind rune)` and call it with `val = replaceAllWithLookbehind(val, ",", "\\,", "\\")`.

Reply:
Thanks for the advice. Extracted as a function in a new commit! 😊